org.eclipse.winery.topologymodeler.ui/TmDependencyTestApp/src/ttopology-template.ts

```diff
@@ -0,0 +1,220 @@
+            }
+            console.log(nodeTemplate);
+        } else if (updatedAttribute === ('minInstances') || updatedAttribute === ('maxInstances')) {
+            if (Number.isNaN(+updatedValue)) {
+                nodeTemplate[updatedAttribute] = updatedValue;
+            } else {
+                nodeTemplate[updatedAttribute] = +updatedValue;
+            }
+        } else {
+            console.log(updatedValue);
+            nodeTemplate[updatedAttribute] = updatedValue;
+            console.log(nodeTemplate);
+        }
+        return nodeTemplate;
+    }
+
+    public get state(): DifferenceStates {
+        return this._state;
+    }
+
+    public set state(value: DifferenceStates) {
+        this._state = value;
+        this.color = VersionUtils.getElementColorByDiffState(value);
+    }
+}
+
+/**
+ * This is the datamodel for the Entity Types
+ */
+export class EntityType {
+    constructor(public id: string,
+                public qName: string,
+                public name: string,
+                public namespace: string,
+                public color?: string,
+                public full?: any) {
+    }
+}
+
+/**
+ * This is the datamodel for relationship templates
+ */
+export class TRelationshipTemplate extends AbstractTTemplate {
+    /*
+     get targetElement(): string {
+     return this.targetElement;
+     }
+     get sourceElement(): string {
+     return this.sourceElement;
+     }
+     */
+    constructor(public sourceElement: { ref: string },
+                public targetElement: { ref: string },
+                public name?: string,
+                public id?: string,
+                public type?: string,
+                documentation?: any,
+                any?: any,
+                otherAttributes?: any,
+                public state?: DifferenceStates) {
+        super(documentation, any, otherAttributes);
+    }
+
+    /**
+     * needed for the winery redux reducer,
+     * updates a specific attribute and returns the whole new relationship template
+     * @param indexOfUpdatedAttribute: index of the to be updated attribute in the constructor
+     * @param updatedValue: the new value
+     *
+     * @return relTemplate: a new relationship template with the updated value
+     */
+    generateNewRelTemplateWithUpdatedAttribute(updatedAttribute: string, updatedValue: any): TRelationshipTemplate {
+        const relTemplate = new TRelationshipTemplate(this.sourceElement, this.targetElement, this.name, this.id, this.type,
+            this.documentation, this.any, this.otherAttributes);
+        relTemplate[updatedAttribute] = updatedValue;
+        return relTemplate;
+    }
+
+}
+
+/**
+ * This is the datamodel for the style of nodes and relationships
+ */
+export class Visuals {
+
+    constructor(public color: string,
+                public nodeTypeId: string,
+                public localName?: string,
+                public imageUrl?: string) {
+    }
+
+    /*
+     get color(): string {
```

Comment:
If these methods are not used, remove them.

Reply:
They are used depending on what you want to test, i.e., mock data vs. fetch data